### ISSUE TITLE
WIP: RUCSS times out when large blocks of inline CSS are present in the source code

### ISSUE THREAD

**Issue by NataliaDrause** [wp-media/wp-rocket#4683]
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version - 3.10.6
- Used the search feature to ensure that the bug hasn’t been reported before - yes

**Describe the bug**
This issue is created to monitor the following problem and see number of sites affected by this:

Pages with large inline CSS blocks time out and no Used CSS is added.

The following error is logged in the WP Rocket debug log:
```
Error when contacting the RUCSS API.
cURL error 28: Operation timed out after 5000 milliseconds with 0 bytes received.
```
Screenshot:
![image](https://user-images.githubusercontent.com/60236665/150857021-eea571a3-4fb9-4d29-8b8f-b3d6c24cf2e2.png)

Remove Unused CSS pre-processes external CSS files when we enable the option. The rest of the CSS code that is present on the HTML (inline CSS) is being processed on the fly when cache is generated.

We give it 5 seconds here: https://github.com/wp-media/wp-rocket/blob/15574e3957478c71210c315dd08488778aad5eb5/inc/Engine/Optimization/RUCSS/Frontend/APIClient.php#L33

If Used CSS is not generated within 5 seconds, it times out and doesn't add any Used CSS.

Changing it to a higher value for timeout resolves the issue. But this cannot be used as a workaround since a non-cached page will be hanging for all that time.

A better workaround is to move the inline CSS code to the external stylesheet. This resolves the problem for some sites. But this is specific to the theme/plugin and overall site structure.

**Related tickets:**
https://secure.helpscout.net/conversation/1683501626/305165?folderId=3864740#thread-4941049285
https://secure.helpscout.net/conversation/1759865188/319838/#thread-5192299393

Slack thread: https://wp-media.slack.com/archives/C029G1B5HD2/p1636464992083800?thread_ts=1636052705.067800&cid=C029G1B5HD2

**To Reproduce**
Reproduced on customers sites only:
1. Have a page with very large blocks of inline CSS.
2. Enable RUCSS and wait for success message.
3. Clear WP Rocket cache just to eliminate a possibility that Used CSS is hiding behind cache and also to see cache generation in progress.
4. Visit the page in the incognito browser - it will hang for 5 seconds. 
5. Once page is loaded, no Used CSS is added. 
6. WP Rocket debug would log the error as shared above.

Note: removing the large block of inline CSS resolves the problem.

**Expected behavior**
Used CSS should be generated within 5 sec timeframe.

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

**Comment by jazir555** [comment]
Yeah if you could fix the unused css generation hang time when first loading a page that would be very helpful, this makes it unusable if you are developing a site and have to clear the cache frequently. Terrible user experience if they encounter that. They'll leave before the unused css is even generated.

**Comment by piotrbak** [comment]
Fixed with 3.11
